src/main/php/PHPMD/AbstractNode.php

```diff
@@ -310,7 +310,7 @@ public function getMetric($name)
      */
     public function setMetrics(array $metrics): void
     {
-        if ($this->metrics === null) {
+        if (!isset($this->metrics)) {
             $this->metrics = $metrics;
         }
```

Comment:
```suggestion
        $this->metrics ??= $metrics;
```

I'm also now thinking that we should probably throw something if `->setMetrics()` is called when `$this->metrics` is already set instead of just doing nothing silently.

Reply:
```suggestion
        if (isset($this->metrics)) {
            throw new LogicException('Metrics cannot be overridden');
        }
```